feature/policy_forwarding/GUE/Outer_IPv4/decapsulation/otg_tests/Inner_IP/README.md

```diff
@@ -0,0 +1,337 @@
+# PF-1.4: Interface based GUE Decapsulation to IPv4 tunnel
+
+## Summary
+
+This is to test the the functionality of policy-based forwarding (PF) to decapsulate Generic UDP Encapsulation (GUE) traffic. These tests verify the use case of IPv4, and IPv6 encapsulated traffic in IPv4 GUE tuennel. The tests are meant for `Tunnel Interface` or `Policy Based` implementation of IPv4 GUE tunnel. The tests validate that the DUT performs the following action.
+
+ - Decapsulate the outer (transport) layer 3 and GUE headers of GUE packets destined to locally-configured decap VIP and matching UDP port after that it will forward them based on the exposed inner (payload) L3 header.
+ - GUE Inner protocol type must be derived from a unique DST port. If not specifically configured, then the following default DST UDP port will be used.
+    - For inner IPv4 - GUE UDP port 6080
+    - For inner IPv6 - GUE UDP port 6615
+ - Post decapsulation the DUT should copy outer TTL(and decrement) to inner header and maintain the inner DSCP vaule.
```

Comment:
" copy outer TTL(and decrement) to inner header and maintain the inner DSCP value"

- For TTL, requirement is to copy inner TTL to outer during Encap. However, it isnt a requirement to do the reverse during Decap.
- Maintaining the inner DSCP value during Decap isnt a requirement anymore.

Reply:
if you don't copy the ttl back post decapsulation, then what will ttl look like.

For example:

---ip4-ttl-30----R1(encap)----R2----R3-----R4----R5(decap)----->do we expect ip4 ttl25 here or not?

For DSCP, Cisco Fretta and 8K have different implementation.

Currently, if I understand correctly, then inner DSCP is not modified. At the encapsulation time, we can match dscp and modify outer tunnel dscp header, but at the decapsulation time, the inner dscp remains as is.